### PR TITLE
ci: add zizmor and harden workflows against its findings

### PR DESCRIPTION
**Pull request by kazupon**
## Summary

Fix zizmor errors

## Risk

- Risk level: low / medium / high
- User or production impact:
- Rollback plan:

## Validation

- [ ] Automated tests or checks run:
- [ ] Manual verification completed:
- [ ] Edge cases or failure modes considered:

## Release and docs checklist

- [ ] Documentation, comments, or runbooks updated, or not needed.
- [x] Configuration, migration, or environment changes documented, or not needed.
- [ ] Observability, logging, and alerting impact considered, or not needed.
- [ ] Security, privacy, and dependency impact considered, or not needed.

### CODE REVIEW

File: .github/workflows/publish.yml [hunk 1]
```diff
@@ -377,12 +377,13 @@ jobs:
...
         with:
           fetch-depth: 0
           persist-credentials: false
 
       - name: Create Release
-        uses: softprops/action-gh-release@v3
-        with:
-          generate_release_notes: true
+        env:
+          GH_TOKEN: ${{ github.token }}
```
Comment (ubugeeei) [inline .github/workflows/publish.yml:387]: can we remove token?

Comment (kazupon) [reply .github/workflows/publish.yml:387]: `GH_TOKEN` is required: the `gh` CLI authenticates from `GH_TOKEN` (or `GITHUB_TOKEN`) in the step env, and neither is injected automatically by the runner.

The git-remote fallback is also closed because `actions/checkout` uses `persist-credentials: false`.

Without it, `gh release create` fails with an auth error.

FYI:
The previous `softprops/action-gh-release@v3` didn't need an explicit token only because the action defaults its `token:` input to `${{ github.token }}` internally, the plain `gh` CLI has no such implicit lookup.

**Review by ubugeeei — APPROVED** [review]
